Generated conanfile.py should comply with PEP8 and be friendly to static code analyzers

As of 1.1.1, `conanfile.py` for the main and test packages automatically generated through `conan new -t foo/0.1@username/testing` don't comply with PEP8. This is a bit annoying if you care about style to have to fix the same errors again, or avoid code generation features.

Here's the current state:
```
$ pip install --user pycodestyle
$ pycodestyle conanfile.py test_package/conanfile.py 
conanfile.py:18:80: E501 line too long (85 > 79 characters)
conanfile.py:30:80: E501 line too long (82 > 79 characters)
test_package/conanfile.py:4:1: E302 expected 2 blank lines, found 1
test_package/conanfile.py:10:80: E501 line too long (96 > 79 characters)
```

3 out of 4 warnings are about [maximum line length](https://www.python.org/dev/peps/pep-0008/#maximum-line-length). The 4th is a missing blank line.

Another one that is not mentionned by the `pycodestyle` tool but is good practice is about the [imports](https://www.python.org/dev/peps/pep-0008/#imports). The order is wrong, and import groups should be separated by a newline:

**bad**
```
from conans import ConanFile, CMake, tools
import os
```

**good**
```
import os

from conans import ConanFile, CMake, tools
```

As an additional style remark (unrelated to PEP8), I find this code in the test_package to have poor readability:
```
self.run(".%sexample" % os.sep)
```

```
import os.path
...
self.run(os.path.join('.', 'example'))
```
A `git grep -w example` will fail on the first version, as the formatting is mixed with the content.

Seen on Conan 1.1.1 on Ubuntu 14.04

- [X] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [X] I've specified the Conan version, operating system version and any tool that can be relevant.
- [X] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

--- a/conans/client/cmd/new.py
+++ b/conans/client/cmd/new.py
@@ -21,9 +21,11 @@ class {package_name}Conan(ConanFile):
     def source(self):
         self.run("git clone https://github.com/memsharded/hello.git")
         self.run("cd hello && git checkout static_shared")
-        # This small hack might be useful to guarantee proper /MT /MD linkage in MSVC
-        # if the packaged project doesn't have variables to set it properly
-        tools.replace_in_file("hello/CMakeLists.txt", "PROJECT(MyHello)", '''PROJECT(MyHello)
+        # This small hack might be useful to guarantee proper /MT /MD linkage
+        # in MSVC if the packaged project doesn't have variables to set it
+        # properly
+        tools.replace_in_file("hello/CMakeLists.txt", "PROJECT(MyHello)",
+                              '''PROJECT(MyHello)
 include(${{CMAKE_BINARY_DIR}}/conanbuildinfo.cmake)
 conan_basic_setup()''')
 
@@ -33,7 +35,8 @@ conan_basic_setup()''')
         cmake.build()
 
         # Explicit way:
-        # self.run('cmake %s/hello %s' % (self.source_folder, cmake.command_line))
+        # self.run('cmake %s/hello %s'
+        #          % (self.source_folder, cmake.command_line))
         # self.run("cmake --build . %s" % cmake.build_config)
 
     def package(self):
@@ -46,10 +49,11 @@ conan_basic_setup()''')
 
     def package_info(self):
         self.cpp_info.libs = ["hello"]
+
 """
 
-conanfile_bare = """from conans import ConanFile
-from conans import tools
+conanfile_bare = """from conans import ConanFile, tools
+
 
 class {package_name}Conan(ConanFile):
     name = "{name}"
@@ -87,7 +91,8 @@ class {package_name}Conan(ConanFile):
         cmake.build()
 
         # Explicit way:
-        # self.run('cmake %s/src %s' % (self.source_folder, cmake.command_line))
+        # self.run('cmake %s/hello %s'
+        #          % (self.source_folder, cmake.command_line))
         # self.run("cmake --build . %s" % cmake.build_config)
 
     def package(self):
@@ -102,8 +107,9 @@ class {package_name}Conan(ConanFile):
         self.cpp_info.libs = ["hello"]
 """
 
-conanfile_header = """from conans import ConanFile, tools
-import os
+conanfile_header = """import os
+
+from conans import ConanFile, tools
 
 
 class {package_name}Conan(ConanFile):
@@ -116,20 +122,23 @@ class {package_name}Conan(ConanFile):
     # No settings/options are necessary, this is header only
 
     def source(self):
-        '''retrieval of the source code here. Remember you can also put the code in the folder and
-        use exports instead of retrieving it with this source() method
+        '''retrieval of the source code here. Remember you can also put the code
+        in the folder and use exports instead of retrieving it with this
+        source() method
         '''
-        #self.run("git clone ...") or
-        #tools.download("url", "file.zip")
-        #tools.unzip("file.zip" )
+        # self.run("git clone ...") or
+        # tools.download("url", "file.zip")
+        # tools.unzip("file.zip" )
 
     def package(self):
         self.copy("*.h", "include")
 """
 
 
-test_conanfile = """from conans import ConanFile, CMake, tools
-import os
+test_conanfile = """import os
+
+from conans import ConanFile, CMake, tools
+
 
 class {package_name}TestConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
@@ -137,7 +146,8 @@ class {package_name}TestConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
-        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is in "test_package"
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
         cmake.configure()
         cmake.build()
 
@@ -258,7 +268,7 @@ def cmd_new(ref, header=False, pure_c=False, test=False, exports_sources=False, 
         files = {"conanfile.py": conanfile.format(name=name, version=version,
                                                   package_name=package_name)}
         if pure_c:
-            config = "\n    def configure(self):\n        del self.settings.compiler.libcxx"
+            config = "    def configure(self):\n        del self.settings.compiler.libcxx\n"
             files["conanfile.py"] = files["conanfile.py"] + config
 
     if test:

--- a/conans/model/conan_file.py
+++ b/conans/model/conan_file.py
@@ -240,9 +240,15 @@ class ConanFile(object):
         """
 
     def build(self):
+        """ build your project calling the desired build tools as done in the command line.
+        E.g. self.run("cmake --build .") Or use the provided build helpers. E.g. cmake.build()
+        """
         self.output.warn("This conanfile has no build step")
 
     def package(self):
+        """ package the needed files from source and build folders.
+        E.g. self.copy("*.h", src="src/includes", dst="includes")
+        """
         self.output.warn("This conanfile has no package step")
 
     def package_info(self):
@@ -267,6 +273,9 @@ class ConanFile(object):
         """
 
     def test(self):
+        """ test the generated executable.
+        E.g.  self.run("./example")
+        """
         raise ConanException("You need to create a method 'test' in your test/conanfile.py")
 
     def __repr__(self):

